matchzoo/models/match_lstm.py

```diff
@@ -0,0 +1,83 @@
+"""Match LSTM model."""
+import keras
+import keras.backend as K
+
+from matchzoo import engine
+
+
+class MatchLSTM(engine.BaseModel):
+    """
+    Match LSTM model.
+
+    Examples:
+        >>> model = MatchLSTM()
+        >>> model.guess_and_fill_missing_params(verbose=0)
+        >>> model.build()
+        >>> model.params['fc_hidden_size'] = 200
+        >>> model.params['rnn_hidden_size'] = 256
+
+    """
+
+    @classmethod
+    def get_default_params(cls):
+        """Get default parameters."""
+        params = super().get_default_params(with_embedding=True)
+        params.add(engine.Param(
+            'rnn_hidden_size', 256,
+            hyper_space=engine.hyper_spaces.quniform(low=128, high=384, q=128)
+        ))
+        params.add(engine.Param(
+            'fc_hidden_size', 200,
+            hyper_space=engine.hyper_spaces.quniform(
+                low=100, high=300, q=100)
+        ))
+        return params
+
+    def build(self):
+        """Build model."""
+        query, doc = self._make_inputs()
+        self.query_len = query.shape[1]
```

Comment:
Do not store things in `self`. Change the signature of your attention method to take those as parameters.

Reply:
When a tenor is passed through  LSTM layer in Keras, the seq_length dimension of the output tensor is None. And the attention function only can receive tensor as its parameter, so I change the attention as an inner function of build to get get the length information.

Reply:
@lixinsu Yep that is a better way. An alternative is to make it a `classmethod`.